### ISSUE TITLE
Import dask_ml with python 3.10 failed due to conflict with dask.distributed

### ISSUE THREAD

**Issue by hungbui-io** [dask/dask-ml#969]
**Describe the issue**:
Can not import dask_ml with python 3.10.10, it shows error: `ImportError: cannot import name 'Future' from 'dask.distributed'`
It seems like dask-ml latest version is not compatible with dask.distributed.
**Minimal Complete Verifiable Example**:

```python
Python 3.10.10 (tags/v3.10.10:aad5f6a, Feb  7 2023, 17:20:36) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import dask_ml
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\...\site-packages\dask_ml\__init__.py", line 4, in <module>
    from dask_ml.model_selection import _normalize
  File "C:\Users\...\site-packages\dask_ml\model_selection\__init__.py", line 6, in <module>
    from ._hyperband import HyperbandSearchCV
  File "C:\Users\...\site-packages\dask_ml\model_selection\_hyperband.py", line 12, in <module>
    from ._incremental import BaseIncrementalSearchCV
  File "C:\Users\...\site-packages\dask_ml\model_selection\_incremental.py", line 19, in <module>
    from dask.distributed import Future, default_client, futures_of, wait
ImportError: cannot import name 'Future' from 'dask.distributed' (C:\Users\...\site-packages\dask\distributed.py)
```

**Environment**:

- Dask version: dask-ml: 2023.3.24, dask:2023.3.2
- Python version: 3.10.10
- Operating System: Windows 10
- Install method (conda, pip, source): pip

**Comment by hungbui-io** [comment]
@TomAugspurger In my environment, I am able to import distributed and dask.distributed, but fail to import Future from dask.distributed.
```python
Python 3.10.10 (tags/v3.10.10:aad5f6a, Feb  7 2023, 17:20:36) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import dask.distributed
>>> import distributed
>>> from dask.distributed import Future
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'Future' from 'dask.distributed' (C:\Users\...\site-packages\dask\distributed.py)
```

**Comment by hungbui-io** [comment]
@TomAugspurger  I aslo try with a fresh new virtual env, but the error is still there. Only differences between your system and my system is that I use Windows 10 and python 3.10.10
```
(old-venv) Administrator@EC2AMAZ-J1SSAH8 C:\Users\...\site-packages
# python -m venv dl

(old-venv) Administrator@EC2AMAZ-J1SSAH8 C:\Users\...\site-packages
# dl\Scripts\activate

(dl) Administrator@EC2AMAZ-J1SSAH8 C:\Users\...\site-packages
# pip list
Package    Version
---------- -------
pip        22.3.1
setuptools 65.5.0

[notice] A new release of pip available: 22.3.1 -> 23.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip

(dl) Administrator@EC2AMAZ-J1SSAH8 C:\Users\...\site-packages
# pip install dask-ml==2023.3.24
Collecting dask-ml==2023.3.24
  Downloading dask_ml-2023.3.24-py3-none-any.whl (148 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 148.7/148.7 kB 8.7 MB/s eta 0:00:00
Collecting scikit-learn>=1.2.0
  Downloading scikit_learn-1.2.2-cp310-cp310-win_amd64.whl (8.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 8.3/8.3 MB 66.4 MB/s eta 0:00:00
Collecting numpy>=1.20.0
  Downloading numpy-1.24.2-cp310-cp310-win_amd64.whl (14.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 14.8/14.8 MB 54.7 MB/s eta 0:00:00
Collecting multipledispatch>=0.4.9
  Using cached multipledispatch-0.6.0-py3-none-any.whl (11 kB)
Collecting dask-glm>=0.2.0
  Using cached dask_glm-0.2.0-py2.py3-none-any.whl (12 kB)
Collecting packaging
  Using cached packaging-23.0-py3-none-any.whl (42 kB)
Collecting scipy
  Using cached scipy-1.10.1-cp310-cp310-win_amd64.whl (42.5 MB)
Collecting dask[array,dataframe]>=2.4.0
  Using cached dask-2023.3.2-py3-none-any.whl (1.2 MB)
Collecting distributed>=2.4.0
  Using cached distributed-2023.3.2.1-py3-none-any.whl (957 kB)
Collecting pandas>=0.24.2
  Downloading pandas-2.0.0-cp310-cp310-win_amd64.whl (11.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 11.2/11.2 MB 72.5 MB/s eta 0:00:00
Collecting numba>=0.51.0
  Using cached numba-0.56.4-cp310-cp310-win_amd64.whl (2.5 MB)
Collecting cloudpickle>=0.2.2
  Using cached cloudpickle-2.2.1-py3-none-any.whl (25 kB)
Collecting fsspec>=0.6.0
  Using cached fsspec-2023.3.0-py3-none-any.whl (145 kB)
Collecting click>=7.0
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting toolz>=0.8.2
  Using cached toolz-0.12.0-py3-none-any.whl (55 kB)
Collecting importlib-metadata>=4.13.0
  Using cached importlib_metadata-6.1.0-py3-none-any.whl (21 kB)
Collecting partd>=1.2.0
  Using cached partd-1.3.0-py3-none-any.whl (18 kB)
Collecting pyyaml>=5.3.1
  Downloading PyYAML-6.0-cp310-cp310-win_amd64.whl (151 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 151.7/151.7 kB 9.4 MB/s eta 0:00:00
Collecting zict>=2.1.0
  Using cached zict-2.2.0-py2.py3-none-any.whl (23 kB)
Collecting tornado>=6.0.3
  Using cached tornado-6.2-cp37-abi3-win_amd64.whl (425 kB)
Collecting psutil>=5.7.0
  Downloading psutil-5.9.4-cp36-abi3-win_amd64.whl (252 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 252.5/252.5 kB ? eta 0:00:00
Collecting tblib>=1.6.0
  Using cached tblib-1.7.0-py2.py3-none-any.whl (12 kB)
Collecting urllib3>=1.24.3
  Using cached urllib3-1.26.15-py2.py3-none-any.whl (140 kB)
Collecting jinja2>=2.10.3
  Using cached Jinja2-3.1.2-py3-none-any.whl (133 kB)
Collecting sortedcontainers>=2.0.5
  Using cached sortedcontainers-2.4.0-py2.py3-none-any.whl (29 kB)
Collecting msgpack>=1.0.0
  Using cached msgpack-1.0.5-cp310-cp310-win_amd64.whl (61 kB)
Collecting locket>=1.0.0
  Using cached locket-1.0.0-py2.py3-none-any.whl (4.4 kB)
Collecting six
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting numpy>=1.20.0
  Using cached numpy-1.23.5-cp310-cp310-win_amd64.whl (14.6 MB)
Requirement already satisfied: setuptools in c:\users\...\site-packages (from numba>=0.51.0->dask-ml==2023.3.24) (65.5.0)
Collecting llvmlite<0.40,>=0.39.0dev0
  Using cached llvmlite-0.39.1-cp310-cp310-win_amd64.whl (23.2 MB)
Collecting pytz>=2020.1
  Downloading pytz-2023.3-py2.py3-none-any.whl (502 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 502.3/502.3 kB 32.8 MB/s eta 0:00:00
Collecting tzdata>=2022.1
  Using cached tzdata-2023.3-py2.py3-none-any.whl (341 kB)
Collecting python-dateutil>=2.8.2
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting joblib>=1.1.1
  Using cached joblib-1.2.0-py3-none-any.whl (297 kB)
Collecting threadpoolctl>=2.0.0
  Using cached threadpoolctl-3.1.0-py3-none-any.whl (14 kB)
Collecting colorama
  Downloading colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Collecting zipp>=0.5
  Using cached zipp-3.15.0-py3-none-any.whl (6.8 kB)
Collecting MarkupSafe>=2.0
  Using cached MarkupSafe-2.1.2-cp310-cp310-win_amd64.whl (16 kB)
Collecting heapdict
  Using cached HeapDict-1.0.1-py3-none-any.whl (3.9 kB)
Installing collected packages: sortedcontainers, pytz, msgpack, heapdict, zipp, zict, urllib3, tzdata, tornado, toolz, threadpoolctl, tblib, six, pyyaml, psutil, packaging, numpy, MarkupSafe, locket, llvmlite, joblib, fsspec, colorama, cloudpickle, scipy, python-dateutil, partd, numba, multipledispatch, jinja2, importlib-metadata, click, scikit-learn, pandas, dask, distributed, dask-glm, dask-ml
Successfully installed MarkupSafe-2.1.2 click-8.1.3 cloudpickle-2.2.1 colorama-0.4.6 dask-2023.3.2 dask-glm-0.2.0 dask-ml-2023.3.24 distributed-2023.3.2.1 fsspec-2023.3.0 heapdict-1.0.1 importlib-metadata-6.1.0 jinja2-3.1.2 joblib-1.2.0 llvmlite-0.39.1 locket-1.0.0 msgpack-1.0.5 multipledispatch-0.6.0 numba-0.56.4 numpy-1.23.5 packaging-23.0 pandas-2.0.0 partd-1.3.0 psutil-5.9.4 python-dateutil-2.8.2 pytz-2023.3 pyyaml-6.0 scikit-learn-1.2.2 scipy-1.10.1 six-1.16.0 sortedcontainers-2.4.0 tblib-1.7.0 threadpoolctl-3.1.0 toolz-0.12.0 tornado-6.2 tzdata-2023.3 urllib3-1.26.15 zict-2.2.0 zipp-3.15.0

[notice] A new release of pip available: 22.3.1 -> 23.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip

(dl) Administrator@EC2AMAZ-J1SSAH8 C:\Users\...\site-packages
# python
Python 3.10.10 (tags/v3.10.10:aad5f6a, Feb  7 2023, 17:20:36) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import dask_ml
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\...\site-packages\dask_ml\__init__.py", line 4, in <module>
    from dask_ml.model_selection import _normalize
  File "C:\Users\...\site-packages\dask_ml\model_selection\__init__.py", line 6, in <module>
    from ._hyperband import HyperbandSearchCV
  File "C:\Users\...\site-packages\dask_ml\model_selection\_hyperband.py", line 12, in <module>
    from ._incremental import BaseIncrementalSearchCV
  File "C:\Users\...\site-packages\dask_ml\model_selection\_incremental.py", line 19, in <module>
    from dask.distributed import Future, default_client, futures_of, wait
ImportError: cannot import name 'Future' from 'dask.distributed' (C:\Users\...\site-packages\dask\distributed.py)
>>>

```

**Comment by hungbui-io** [comment]
Hi @TomAugspurger, I found the root cause of the issue There is an issue with my Python installation iteself. Reinstalling Python solved the issue. I am closing this issue. 
Thanks a lot for your support.
